contrib/benchmarkDictBuilder/benchmark.c

```diff
@@ -0,0 +1,458 @@
+    goto _cleanup;
+  }
+  DISPLAYLEVEL(2, "RANDOM took %f seconds to execute \n", time_taken);
+
+  double cRatio = compressWithDict(srcInfo, dInfo, randomParam->zParams.compressionLevel, displayLevel);
+  if (cRatio < 0) {
+    DISPLAYLEVEL(1, "Compressing with RANDOM dictionary does not work\n");
+    result = 1;
+    goto _cleanup;
+  }
+  DISPLAYLEVEL(2, "Compression ratio with random dictionary is %f\n", cRatio);
+
+
+_cleanup:
+  freeDictInfo(dInfo);
+  return result;
+}
+
+/** benchmarkCover() :
+ *  Measure how long random dictionary builder takes and compression ratio with the cover dictionary
+ *  @return 0 if benchmark successfully, 1 otherwise
+ */
+int benchmarkCover(sampleInfo *srcInfo, unsigned maxDictSize,
+                ZDICT_cover_params_t *coverParam) {
+  const int displayLevel = coverParam->zParams.notificationLevel;
+  int result = 0;
+  clock_t t;
+  t = clock();
+  dictInfo* dInfo = createDictFromFiles(srcInfo, maxDictSize, NULL, coverParam, NULL);
+  t = clock() - t;
+  double time_taken = ((double)t)/CLOCKS_PER_SEC;
+  if (!dInfo) {
+    DISPLAYLEVEL(1, "COVER does not train successfully\n");
+    result = 1;
+    goto _cleanup;
+  }
+  DISPLAYLEVEL(2, "COVER took %f seconds to execute \n", time_taken);
+
+  double cRatio = compressWithDict(srcInfo, dInfo, coverParam->zParams.compressionLevel, displayLevel);
+  if (cRatio < 0) {
+    DISPLAYLEVEL(1, "Compressing with COVER dictionary does not work\n");
+    result = 1;
+    goto _cleanup;
+  }
+  DISPLAYLEVEL(2, "Compression ratio with cover dictionary is %f\n", cRatio);
+
+_cleanup:
+  freeDictInfo(dInfo);
+  return result;
+}
+
+
+
+/** benchmarkLegacy() :
+ *  Measure how long legacy dictionary builder takes and compression ratio with the legacy dictionary
+ *  @return 0 if benchmark successfully, 1 otherwise
+ */
+int benchmarkLegacy(sampleInfo *srcInfo, unsigned maxDictSize, ZDICT_legacy_params_t *legacyParam) {
+  const int displayLevel = legacyParam->zParams.notificationLevel;
+  int result = 0;
+  clock_t t;
+  t = clock();
+  dictInfo* dInfo = createDictFromFiles(srcInfo, maxDictSize, NULL, NULL, legacyParam);
+  t = clock() - t;
+  double time_taken = ((double)t)/CLOCKS_PER_SEC;
+  if (!dInfo) {
+    DISPLAYLEVEL(1, "LEGACY does not train successfully\n");
+    result = 1;
+    goto _cleanup;
+
+  }
+  DISPLAYLEVEL(2, "LEGACY took %f seconds to execute \n", time_taken);
+
+  double cRatio = compressWithDict(srcInfo, dInfo, legacyParam->zParams.compressionLevel, displayLevel);
+  if (cRatio < 0) {
+    DISPLAYLEVEL(1, "Compressing with LEGACY dictionary does not work\n");
+    result = 1;
+    goto _cleanup;
+
+  }
+  DISPLAYLEVEL(2, "Compression ratio with legacy dictionary is %f\n", cRatio);
+
+_cleanup:
+  freeDictInfo(dInfo);
+  return result;
+}
+
+
+
+int main(int argCount, const char* argv[])
+{
+  int displayLevel = 2;
+  const char* programName = argv[0];
+  int result = 0;
+  /* Initialize arguments to default values */
+  unsigned k = 200;
```

Comment:
nit: make these `const`, we can always mark them non-const if we allow them to be modified later on.